llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -966,6 +1012,40 @@ Error MCCASBuilder::buildFragments() {
       continue;
     }
 
+    // If the Section is the DWARF Line Section, append all the contents of the
+    // section into a buffer and split it up into multiple CAS blocks, where one
+    // block represents one function's contribution into the section or one line
+    // table.
+    if (Asm.getContext().getObjectFileInfo()->getDwarfLineSection() == &Sec) {
```

Comment:
I understand the debug str handling has been inlined in the middle of the loop, but it likely won't scale very well once all debug sections are accounted for. I suggest outlining the body of the `if` into a helper function.

Reply:
The only reason I wrote it inline was that this function will only be called once, but I do agree with you, it doesn't scale well for every debug information section. I will make a separate function for it, good catch!